src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -409,6 +413,43 @@ final class TrialService
   private def countProjects: Future[PerRequestMessage] =
     Future(RequestComplete(OK, trialDao.countProjects))
 
+  private def adoptProject(projectName: String): Future[PerRequestMessage] = {
+    val project = RawlsBillingProjectName(projectName)
+    val record = Try(trialDao.getProjectRecord(project)) match {
+      case Success(p) => p // project exists!
+      case Failure(ex) =>
+        if (ex.getMessage.contains("not found")) {
+          trialDao.insertProjectRecord(project)
```

Comment:
Does this function result in a new project being created? I assumed based on your comments that `adopt` would get a project had already been created.

Reply:
no new project is created here! This just adds a record into our ES index that references a project. In fact, we don't even check or care if the project exists or is valid - we just insert the reference.